compiler/kernel/op.go

```diff
@@ -578,29 +581,34 @@ func (b *Builder) compileTrunk(trunk *dag.Trunk, parent zbuf.Puller) ([]zbuf.Pul
 		// common "from" source are distributed across the collection
 		// of SequenceScanner operators and the upstream parent distributes work
 		// across the parallel instances of trunks.
-		lister, ok := b.listers[src]
+		slicer, ok := b.slicers[src]
 		if !ok {
 			lk := b.source.Lake()
 			pool, err := lk.OpenPool(b.pctx.Context, src.ID)
 			if err != nil {
 				return nil, err
 			}
-			l, err := meta.NewSortedLister(b.pctx.Context, lk, pool, src.Commit, filter)
+			// We pass a new type context in here because we don't want the metadata types to interfere
+			// with the downstream flowgraph.  And there's no need to map between contexts because
+			// the metadata here is intercepted by the scanner and these zed values never enter
+			// the flowgraph.  For the metaqueries below, we pass in the flowgraph's type context
+			// because this data does, in fact, flow into the downstream flowgraph.
+			l, err := meta.NewSortedLister(b.pctx.Context, b.pctx.Zctx, lk, pool, src.Commit, filter)
```

Comment:
Given the comment, should `b.pctx.Zctx` be `zed.NewContext()`?

Reply:
doh!  yes

Reply:
slicer needs to pick it up too